thriftbp/server_transport.go

```diff
@@ -0,0 +1,48 @@
+package thriftbp
+
+import (
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/go-kit/kit/metrics"
+	"github.com/reddit/baseplate.go/metricsbp"
+)
+
+const meterNameTransportConnCounter = "thrift.connections"
+
+type CountedTServerTransport struct {
+	thrift.TServerTransport
+}
+
+func (m *CountedTServerTransport) Accept() (thrift.TTransport, error) {
+
+	transport, err := m.TServerTransport.Accept()
+	if err != nil {
+		return nil, err
+	}
+
+	return NewCountedTTransport(transport), nil
+}
+
+type CountedTTransport struct {
+	thrift.TTransport
+	counter metrics.Counter
+}
+
+func NewCountedTTransport(transport thrift.TTransport) thrift.TTransport {
```

Comment:
Since this is an exported constructor you should either:
* Return `*CountedTTransport` or
* Unexport `CountedTTransport` and keep returning the interface

I would probably slightly lean toward the latter since you don't expect people outside the package to need or name the type, just the functionality.